server/gateway.go

```diff
@@ -2012,6 +2038,16 @@ func (s *Server) gatewayUpdateSubInterest(accName string, sub *subscription, cha
 		}
 	}
 	if first || last {
+		if first && sub.client != nil {
+			c := sub.client
+			if c.in.rsubs == nil {
+				c.in.rsubs = NewSublist()
+			}
+			c.in.rsubs.Insert(sub)
+			time.AfterFunc(s.gateway.recSubExp, func() {
```

Comment:
This need to be protected at all if server is shutdown in between?

Reply:
Not sure what you mean here.. if in test we shutdown a server and there may be some of those timers still running, they will execute when time comes and remove the closed subscription from their sublist.

Reply:
Mostly thinking embedded cases, don't want a server shutdown (but process persists) to crash the app.

Reply:
It would not.. but I can add a test for that. A client's `gw` (since I will move it there) is created once and never set to nil, same for rsubs, so `c.gw.Remove(sub)` even after server is  shutdown would be ok.

Reply:
(see previous comment about having to leave it in c.in, but still same explication as above remains: object will still be reachable after server is shutdown).

Reply:
ok